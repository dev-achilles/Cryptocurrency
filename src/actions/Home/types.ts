export interface HomeState {
  homeData: Array<any>;
  tokenInfo: Object;
}

export enum HomeEnum {
  GET_HOME_DATA = 'GET_HOME_DATA',
  SET_HOME_DATA = 'SET_HOME_DATA',
}

export enum InfoEnum {
  GET_INFO_DATA = 'GET_INFO_DATA',
}

export interface getHomeDataAction {
  type: HomeEnum.GET_HOME_DATA;
  payload: Array<any>;
}

export interface setHomeDataAction {
  type: HomeEnum.SET_HOME_DATA;
  payload: Array<any>;
}

export interface getInfoDataAction {
  type: InfoEnum.GET_INFO_DATA;
  payload: any;
}

export type CryptocurrenciesType = {
  name: string;
  category: string;
  goal: string;
  interest: string;
  dateActive: string;
  column: string;
  date: string;
  important: string;
  ticker: string;
  tokenType: string;
  tokenPrice: string;
  fundraisingGoal: string;
  totalTokens: string;
  tokenSale: string;
  roleOfToken: string;
  links: Array<string>;
  images: {
    logo: string;
    img: string;
    video: string;
  };
};

export type HomeAction = getHomeDataAction | setHomeDataAction | getInfoDataAction;
