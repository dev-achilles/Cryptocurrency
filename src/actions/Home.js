import { GET_HOME_DATA, SET_HOME_DATA } from '../constants';

export const getHomeData = (data) => {
  return {
    type: GET_HOME_DATA,
    payload: data,
  };
};

export const setHomeData = (data) => {
  return{
    type: SET_HOME_DATA,
    payload: data,
  }
}
