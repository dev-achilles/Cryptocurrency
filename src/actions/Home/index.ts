import {
  HomeEnum,
  getHomeDataAction,
  setHomeDataAction,
  InfoEnum,
  getInfoDataAction,
} from './types';

export const HomeActionCreator = {
  getHomeData: (data: Array<any>): getHomeDataAction => ({
    type: HomeEnum.GET_HOME_DATA,
    payload: data,
  }),
  setHomeData: (data: Array<any>): setHomeDataAction => ({
    type: HomeEnum.SET_HOME_DATA,
    payload: data,
  }),
  getInfoData: (info: string, data: Array<any>): getInfoDataAction => ({
    type: InfoEnum.GET_INFO_DATA,
    payload: { info, data },
  }),
};
