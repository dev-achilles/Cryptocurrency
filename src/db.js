const data = {
  cryptocurrency: [
    {
      name: 'MyCryptoCheckout',
      category: 'Payments',
      goal: '$240,000',
      interest: 'Sponsored',
      dateActive: '3d left',
      column: 'none',
      date: '2021-07-26T23:35:00',
    },
    {
      name: 'Coin98 Labs',
      category: 'Platform',
      goal: '$3,750,000',
      interest: 'Not Rated',
      dateActive: '4d left',
      column: 'none',
      date: '2021-07-29T13:35:00',
    },
    {
      name: 'Hi',
      category: 'Blockchain Service',
      goal: 'NOT SET',
      interest: 'Not Rated',
      dateActive: 'TBA',
      column: 'none',
      date: '2021-07-27T13:35:00',
    },
    {
      name: 'Altair',
      category: 'Blockchain Service',
      goal: 'NOT SET',
      interest: 'Not Rated',
      dateActive: 'TBA',
      column: 'none',
      date: '2021-07-20T13:35:00',
    },
    {
      name: 'PolkaCipher',
      category: 'Blockchain Service',
      goal: '$2,020,000',
      interest: 'Not Rated',
      dateActive: 'in 16h',
      column: 'none',
      date: '2021-07-26T08:35:00',
    },
    {
      name: 'Scaleswap',
      category: 'Blockchain Service',
      goal: '$2,500,000',
      interest: 'Not Rated',
      dateActive: 'in 41h',
      column: 'none',
      date: '2021-07-30T13:35:00',
    },
    {
      name: 'Vent Finance',
      category: 'DeFi',
      goal: '$1,020,000',
      interest: 'Not Rated',
      dateActive: '22 Jul',
      column: 'none',
      date: '2021-07-19T13:35:00',
    },
    {
      name: 'DeRace',
      category: 'Blockchain Service',
      goal: '1,700,000',
      interest: 'Not Rated',
      dateActive: '26 Jul',
      column: 'none',
      date: '2021-06-19T17:35:00',
    },
    {
      name: 'CryptoArt.Ai',
      category: 'Marketplace',
      goal: '$1,120,000',
      interest: 'Not Rated',
      dateActive: '18 Jul',
      column: 'none',
      date: '2021-07-26T15:35:00',
    },
    {
      name: 'Glimpse',
      category: 'Marketplace',
      goal: '$1,300,000',
      interest: 'Not Rated',
      dateActive: '16 Jul',
      column: 'none',
      date: '2021-07-26T11:35:00',
    },
    {
      name: 'Omni',
      category: 'Dapp',
      goal: '$200,000',
      interest: 'Not Rated',
      dateActive: '16 Jul',
      column: 'none',
      date: '2021-08-28T22:35:00',
    },
    {
      name: 'YOLOrekt',
      category: 'Blockchain Service',
      goal: '$320,000',
      interest: 'Not Rated',
      dateActive: '16 Jul',
      column: 'none',
      date: '2021-07-26T13:35:00',
    },
  ],
  users: [
    { name: 'Yevhenii', email: 'yevhenii@gmail.com', password: '1234', role: 'admin' },
    { name: 'Roman', email: 'roman@gmail.com', password: '12345', role: 'admin' },
    { name: 'Vladimir', email: 'vova@gmail.com', password: '123456', role: 'user' },
  ],
};

export default data;
