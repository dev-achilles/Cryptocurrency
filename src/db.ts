const data: Object = {
  cryptocurrency: [
    {
      name: 'MyCryptoCheckout',
      category: 'Payments',
      goal: '$240,000',
      interest: 'Sponsored',
      dateActive: '3d left',
      column: 'none',
      date: '2021-08-17T23:35:00',
      important:
        'MyCryptoCheckout will hold a Genesis event on the 14. Aug. The Genesis will happen on CoinList.',
      ticker: 'RWD',
      tokenType: 'ERC20',
      tokenPrice: '1 RWD = 0.012 USD',
      fundraisingGoal: '75,000,000 TOKEN',
      totalTokens: '500,000,000',
      tokenSale: '50%',
      roleOfToken: 'UTILITY',
      links: [
        {
          name: 'Affiliate Program',
          link: 'https://mycryptocheckout.com/affiliates/',
        },
        {
          name: 'IDO Presale Info',
          link: 'https://mycryptocheckout.com/rwd/',
        },
      ],
      images: {
        logo: 'https://icodrops.com/wp-content/uploads/2021/06/MyCryptoCheckout_logo-150x150.jpeg',
        img: '',
        video: 'https://www.youtube.com/embed/nUoJ9ziaAJQ',
      },
    },

    {
      name: 'Coin98 Labs',
      category: 'Platform',
      goal: '$3,750,000',
      interest: 'Not Rated',
      dateActive: '4d left',
      column: 'none',
      date: '2021-08-17T09:35:00',
      important: 'Coin98 Labs will hold a IEO on the 16. Aug. The IEO will happen on Binance.',
      ticker: 'C98',
      tokenType: 'ERC20',
      tokenPrice: '1 C98 = 0.075 USD',
      fundraisingGoal: '50,000,000 TOKEN',
      totalTokens: '1,000,000,000',
      tokenSale: '50%',
      roleOfToken: 'UTILITY',
      links: [
        {
          name: 'IEO Info',
          link: 'https://launchpad.binance.com/en/subscription/C98_BNB',
        },
      ],
      images: {
        logo: 'https://icodrops.com/wp-content/uploads/2021/07/Coin98Labs_logo-150x150.jpeg',
        img: 'https://icodrops.com/wp-content/uploads/2021/07/xxx-1.png',
        video: '',
      },
    },
    {
      name: 'Hi',
      category: 'Blockchain Service',
      goal: 'NOT SET',
      interest: 'Not Rated',
      dateActive: 'TBA',
      column: 'none',
      date: '2021-08-17T15:35:00',
      important: 'Hi will hold a Pre-Sale on the 30. Jul.',
      ticker: 'HI',
      tokenType: 'ERC20',
      tokenPrice: '1 HI = 0.01 USD',
      fundraisingGoal: 'TOKEN',
      totalTokens: '',
      tokenSale: '',
      roleOfToken: 'UTILITY',
      links: [],
      images: {
        logo: 'https://icodrops.com/wp-content/uploads/2021/07/hi_logo-150x150.png',
        img: '',
        video: 'https://www.youtube.com/embed/f5uiqE5e-1s',
      },
    },
    {
      name: 'Altair',
      category: 'Blockchain Service',
      goal: 'NOT SET',
      interest: 'Not Rated',
      dateActive: 'TBA',
      column: 'none',
      date: '2021-08-17T11:35:00',
      important:
        'Altair will hold a crowdloan and also the distribution through the airdrop on the 21. Sep.',
      ticker: 'AIR',
      tokenType: 'ERC20',
      tokenPrice: '1 AIR = 0.53 USD',
      fundraisingGoal: 'TOKEN',
      totalTokens: '505,000,000',
      tokenSale: '',
      roleOfToken: 'UTILITY',
      links: [
        {
          name: 'Crowdloan Info',
          link: 'https://medium.com/altair-network/the-altair-crowdloan-is-now-live-8b77b0bf7739',
        },
      ],
      images: {
        logo: 'https://icodrops.com/wp-content/uploads/2021/07/Altair_logo-1-150x150.jpeg',
        img: 'https://icodrops.com/wp-content/uploads/2021/07/x-5.png',
        video: '',
      },
    },
    {
      name: 'PolkaCipher',
      category: 'Blockchain Service',
      goal: '$2,020,000',
      interest: 'Not Rated',
      dateActive: 'in 16h',
      column: 'none',
      date: '2021-08-26T09:35:00',
      important:
        'PolkaCipher will hold a Initial Dex Offering on the 25. Jul. The IDO will happen on TrustPad. ',
      ticker: 'CPHR',
      tokenType: 'ERC20',
      tokenPrice: '1 CPHR = 0.017 USD',
      fundraisingGoal: '9,000,000 TOKEN',
      totalTokens: '1,000,000,000',
      tokenSale: '18,9%',
      roleOfToken: 'UTILITY',
      links: [
        { name: 'Tokenomics', link: 'https://docsend.com/view/pu35228mckwdxuv7' },
        {
          name: 'IDO Info',
          link: 'https://polkacipher.medium.com/polkacipher-whitelisting-details-for-trustpad-ido-c0dcc67d9868',
        },
      ],
      images: {
        logo: 'https://icodrops.com/wp-content/uploads/2021/07/PolkaCipher_logo-150x150.jpeg',
        img: 'https://icodrops.com/wp-content/uploads/2021/07/x-11.png',
        video: '',
      },
    },
    {
      name: 'Scaleswap',
      category: 'Blockchain Service',
      goal: '$2,500,000',
      interest: 'Not Rated',
      dateActive: 'in 41h',
      column: 'none',
      date: '2021-08-26T09:35:00',
      important: 'Scaleswap will hold a Initial Dex Offering on the 25. Aug.',
      ticker: 'SCA',
      tokenType: 'ERC20',
      tokenPrice: '1 SCA = 0.35 USD',
      fundraisingGoal: '500,000 TOKEN',
      totalTokens: '25,000,000',
      tokenSale: '46,35%',
      roleOfToken: 'UTILITY',
      links: [
        {
          name: 'Strategic Round',
          link: 'https://markets.businessinsider.com/news/stocks/scaleswap-raises-2-5-million-to-build-layer-2-ido-launchpad-1030446861',
        },
        {
          name: 'Token Info',
          link: 'https://medium.com/scaleswap/introducing-the-scaleswap-sca-token-utility-and-benefits-d9eeb84fe44a',
        },
        {
          name: 'IDO Info',
          link: 'https://medium.com/scaleswap/scaleswap-ido-announcement-shooting-together-for-the-moon-3997fa66870a',
        },
      ],
      images: {
        logo: 'https://icodrops.com/wp-content/uploads/2021/05/Scaleswap_logo-150x150.jpeg',
        img: 'https://icodrops.com/wp-content/uploads/2021/05/%D0%A1%D0%BD%D0%B8%D0%BC%D0%BE%D0%BA-%D1%8D%D0%BA%D1%80%D0%B0%D0%BD%D0%B0-2021-05-20-%D0%B2-17.08.02.png',
        video: '',
      },
    },
    {
      name: 'Vent Finance',
      category: 'DeFi',
      goal: '$1,020,000',
      interest: 'Not Rated',
      dateActive: '22 Jul',
      column: 'none',
      date: '2021-07-19T13:35:00',
      important:
        'Vent Finance will hold a Initial Dex Offering on the 10. Sept. The IDO will happen on VentUp.',
      ticker: 'VENT',
      tokenType: 'ERC20',
      tokenPrice: '',
      fundraisingGoal: 'TOKEN',
      totalTokens: '250,000,000',
      tokenSale: '26,81%',
      roleOfToken: 'UTILITY',
      links: [
        {
          name: 'Pre-Seed Funding Round',
          link: 'https://ventfinance.medium.com/ventup-io-raises-100-000-in-pre-seed-funding-to-launch-an-inclusive-cardano-based-ido-launchpad-dccac4765b4b',
        },
        {
          name: 'SHO Info',
          link: 'https://daomaker.com/company/vent-finance',
        },
      ],
      images: {
        logo: 'https://icodrops.com/wp-content/uploads/2021/06/%D0%A1%D0%BD%D0%B8%D0%BC%D0%BE%D0%BA-%D1%8D%D0%BA%D1%80%D0%B0%D0%BD%D0%B0-2021-07-15-%D0%B2-12.24.54-150x150.png',
        img: 'https://icodrops.com/wp-content/uploads/2021/06/%D0%A1%D0%BD%D0%B8%D0%BC%D0%BE%D0%BA-%D1%8D%D0%BA%D1%80%D0%B0%D0%BD%D0%B0-2021-07-15-%D0%B2-12.26.22.png',
        video: '',
      },
    },
    {
      name: 'DeRace',
      category: 'Blockchain Service',
      goal: '1,700,000',
      interest: 'Not Rated',
      dateActive: '26 Jul',
      column: 'none',
      date: '2021-06-19T17:35:00',
      important:
        'DeRace will hold a Strong Holder Offering on the 26. Jul. The SHO will happen on DAO Maker.',
      ticker: 'DERC',
      tokenType: 'ERC20',
      tokenPrice: '1 DERC = 0.05 USD',
      fundraisingGoal: '7,200,000 TOKEN',
      totalTokens: '120,000,000',
      tokenSale: '34%',
      roleOfToken: 'UTILITY',
      links: [
        {
          name: 'Token Metrics',
          link: 'https://www.derace.io/wp-content/uploads/2021/06/DeRace-Tokenmetrics.pdf',
        },
        {
          name: 'Private Round',
          link: 'https://deracenft.medium.com/derace-closes-an-oversubscribed-private-round-raising-1-7m-55d80101c732',
        },
        {
          name: 'SHO info',
          link: 'https://daomaker.com/company/derace',
        },
      ],
      images: {
        logo: 'https://icodrops.com/wp-content/uploads/2021/07/DeRace_logo-150x150.jpeg',
        img: 'https://icodrops.com/wp-content/uploads/2021/07/x-12.png',
        video: '',
      },
    },
    {
      name: 'CryptoArt.Ai',
      category: 'Marketplace',
      goal: '$1,120,000',
      interest: 'Not Rated',
      dateActive: '18 Jul',
      column: 'none',
      date: '2021-07-26T15:35:00',
      important:
        ' CryptoArt.Ai will hold a Initial Dex Offering on the 18. Jul. The IDO will happen on Poolz.',
      ticker: 'CART',
      tokenType: 'ERC20',
      tokenPrice: '9100 CART = 1 ETH',
      fundraisingGoal: '500,000 TOKEN',
      totalTokens: '',
      tokenSale: '',
      roleOfToken: 'UTILITY',
      links: [],
      images: {
        logo: 'https://icodrops.com/wp-content/uploads/2021/06/CryptoArt.Ai_logo-150x150.jpeg',
        img: 'https://icodrops.com/wp-content/uploads/2021/06/%D0%A1%D0%BD%D0%B8%D0%BC%D0%BE%D0%BA-%D1%8D%D0%BA%D1%80%D0%B0%D0%BD%D0%B0-2021-06-16-%D0%B2-14.29.11.png',
        video: '',
      },
    },
    {
      name: 'Glimpse',
      category: 'Marketplace',
      goal: '$1,300,000',
      interest: 'Not Rated',
      dateActive: '16 Jul',
      column: 'none',
      date: '2021-07-26T11:35:00',
      important:
        'Glimpse will hold a Initial Dex Offering on the 16. Jul. The IDO will happen on BSCPad.',
      ticker: 'GLMS',
      tokenType: 'ERC20',
      tokenPrice: '',
      fundraisingGoal: 'TOKEN',
      totalTokens: '',
      tokenSale: '',
      roleOfToken: 'UTILITY',
      links: [
        {
          name: 'Private Round',
          link: 'https://medium.com/glimpsenft/glimpse-raises-over-1-3m-in-private-round-3bea2b38474e',
        },
        {
          name: 'Bounty Program',
          link: 'https://medium.com/glimpsenft/glimpse-bounty-program-requirements-instructions-how-to-participate-f0ac6c79b68e?source=collection_home---6------1-----------------------',
        },
      ],
      images: {
        logo: 'https://icodrops.com/wp-content/uploads/2021/06/Glimpse_logo-150x150.jpeg',
        img: 'https://icodrops.com/wp-content/uploads/2021/06/z.png',
        video: '',
      },
    },
    {
      name: 'Omni',
      category: 'Dapp',
      goal: '$200,000',
      interest: 'Not Rated',
      dateActive: '16 Jul',
      column: 'none',
      date: '2021-08-28T22:35:00',
      important: 'Omni will hold a Initial Dex Offering on the 16. Jul.',
      ticker: 'OMNI',
      tokenType: 'ERC20',
      tokenPrice: '1 OMNI = 0.02 USD',
      fundraisingGoal: 'TOKEN',
      totalTokens: '',
      tokenSale: '18%',
      roleOfToken: 'UTILITY',
      links: [
        {
          name: 'IDO Info',
          link: 'https://medium.com/omni-app/omni-will-have-anti-bot-protection-during-our-pancake-swap-listing-9c729c1f6214',
        },
      ],
      images: {
        logo: 'https://icodrops.com/wp-content/uploads/2021/03/Omni_logo-150x150.jpg',
        img: 'https://icodrops.com/wp-content/uploads/2021/03/Omni_cover.png',
        video: '',
      },
    },
    {
      name: 'YOLOrekt',
      category: 'Blockchain Service',
      goal: '$320,000',
      interest: 'Not Rated',
      dateActive: '16 Jul',
      column: 'none',
      date: '2021-07-26T13:35:00',
      important: 'YOLOrekt will hold a Liquidity Pool on the 13. Jul.',
      ticker: 'YOLO',
      tokenType: 'ERC20',
      tokenPrice: '',
      fundraisingGoal: 'TOKEN',
      totalTokens: '1,000,000,000',
      tokenSale: '',
      roleOfToken: 'UTILITY',
      links: [
        {
          name: 'Liquditi Pool Info',
          link: 'https://medium.com/yolorekt/the-yolorekt-token-issuance-has-arrived-c1f0e9827840',
        },
      ],
      images: {
        logo: 'https://icodrops.com/wp-content/uploads/2021/07/YOLOrekt_logo-150x150.png',
        img: '',
        video: 'https://player.vimeo.com/video/569561395?dnt=1&app_id=122963&h=afd846de35',
      },
    },
  ],
  users: [
    { name: 'Yevhenii', email: 'yevhenii@gmail.com', password: '1234', role: 'admin' },
    { name: 'Roman', email: 'roman@gmail.com', password: '12345', role: 'admin' },
    { name: 'Vladimir', email: 'vova@gmail.com', password: '123456', role: 'user' },
  ],
};

export default data;
