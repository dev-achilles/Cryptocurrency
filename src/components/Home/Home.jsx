import React, { useState, useEffect } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import IconButton from '@material-ui/core/IconButton';
import Button from '@material-ui/core/Button';
import HighlightOffIcon from '@material-ui/icons/HighlightOff';
import EditIcon from '@material-ui/icons/Edit';
import AddBoxIcon from '@material-ui/icons/AddBox';
import Dialog from '@material-ui/core/Dialog';
import DialogTitle from '@material-ui/core/DialogTitle';
import DialogContent from '@material-ui/core/DialogContent';
import TextField from '@material-ui/core/TextField';
import { getHomeData, setHomeData } from '../../actions/Home';
import { getUser } from '../../actions/User';
import Moment from 'moment';
import { extendMoment } from 'moment-range';
import db from '../../db';

import s from './Home.module.scss';

const moment = extendMoment(Moment);

const useStyles = makeStyles({
  root: {
    borderBottom: '1px solid rgb(119, 118, 118)',
    marginBottom: '10px',
    '&:last-child': {
      marginBottom: '0px',
    },
    '&:hover': {
      backgroundColor: '#fafafa',
      cursor: 'pointer',
    },
  },
  rootActive: {
    borderBottom: '1px solid rgb(119, 118, 118)',
    backgroundColor: '#84ab95',
    marginBottom: '10px',
    '&:last-child': {
      marginBottom: '0px',
    },
  },
  content: {
    '&:last-child': { paddingBottom: '16px' },
    lineHeight: 1.35,
  },
  dialogContent: {
    display: 'flex',
    flexDirection: 'column',
    minWidth: '350px',
    minHeight: '300px',
    justifyContent: 'space-between',
  },
  deleteIcon: {
    color: 'red',
  },
  editIcon: {
    color: 'green',
  },
});

const Home = (props) => {
  const classes = useStyles();
  const data = props.homeData;
  const [active, setActive] = useState(null);
  const [error, setError] = useState(false);
  const [dialog, setDialog] = useState({
    open: false,
    role: '',
    name: '',
  });
  const [dialogValues, setDialogValues] = useState(null);

  useEffect(() => {
    props.dispatch(getHomeData([...db.cryptocurrency]));
  }, []);

  useEffect(() => {
    props.dispatch(getUser());
  }, []);

  const { isLoggedIn } = props.user;
  const setClass = (event) => {
    if (event.target.id !== active) {
      setActive(event.target.id);
    }
    if (event.target.id === 'container') {
      setActive(null);
    }
  };

  const addItemHandle = () => {
    const date = moment(new Date()).format('YYYY-MM-DDTHH:mm');
    setDialog({ open: true, role: 'add', name: '' });
    setDialogValues({ ...dialogValues, ['date']: date });
  };

  const editItemHandle = (name) => {
    setDialog({ open: true, role: 'edit', name: name });
    const items = data.find((item) => item.name === name);
    setDialogValues({ ...items });
  };

  const deleteItemHandle = (name) => {
    data.forEach((item) => {
      if (item.name === name) {
        const items = [...data];
        const index = items.indexOf(item);
        items.splice(index, 1);
        props.dispatch(setHomeData([...items]));
      }
    });
  };

  const handleDialogValueChange = (event) => {
    const { name, value } = event.target;
    setDialogValues({ ...dialogValues, [name]: value });
  };

  const dialogHandle = () => {
    if (dialogValues.name !== undefined && dialogValues.name !== '') {
      setError(false);
      if (dialog.role === 'edit') {
        setDialog({ open: false });
        data.forEach((item) => {
          if (item.name === dialog.name) {
            const itemsData = [...data];
            const index = itemsData.indexOf(item);
            itemsData[index] = { ...dialogValues };
            props.dispatch(setHomeData([...itemsData]));
          }
        });
      }
      if (dialog.role === 'add') {
        const items = [...data];
        props.dispatch(setHomeData([...items, dialogValues]));
      }
      handleDialogClose();
    } else {
      setError(true);
    }
  };

  const handleDialogClose = () => {
    setDialog({ open: false });
    setDialogValues(null);
  };

  const dialogForm = () => {
    return (
      <Dialog
        open={dialog.open}
        onClose={handleDialogClose}
        aria-labelledby="alert-dialog-title"
        aria-describedby="alert-dialog-description">
        <DialogTitle id="alert-dialog-title">Change fields for {dialog.role}</DialogTitle>
        <DialogContent className={classes.dialogContent}>
          <TextField
            label="Name"
            name="name"
            id="outlined-size-small"
            value={dialogValues && dialogValues.name}
            variant="outlined"
            size="small"
            onChange={handleDialogValueChange}
            error={error}
            helperText={error && 'Write please name'}
          />
          <TextField
            label="Category"
            name="category"
            id="outlined-size-small"
            value={dialogValues && dialogValues.category}
            variant="outlined"
            size="small"
            onChange={handleDialogValueChange}
          />
          <TextField
            label="Goal"
            name="goal"
            id="outlined-size-small"
            value={dialogValues && dialogValues.goal}
            variant="outlined"
            size="small"
            onChange={handleDialogValueChange}
          />
          <TextField
            label="Interest"
            name="interest"
            id="outlined-size-small"
            value={dialogValues && dialogValues.interest}
            variant="outlined"
            size="small"
            onChange={handleDialogValueChange}
          />
          <TextField
            id="datetime-local"
            label="Next appointment"
            type="datetime-local"
            variant="outlined"
            size="small"
            name="date"
            value={dialogValues && dialogValues.date}
            onChange={handleDialogValueChange}
          />
        </DialogContent>
        <div className={s.dialog_buttons}>
          <Button
            aria-controls="customized-menu"
            aria-haspopup="true"
            variant="contained"
            color="primary"
            onClick={handleDialogClose}>
            Close
          </Button>
          <Button
            aria-controls="customized-menu"
            aria-haspopup="true"
            variant="contained"
            color="primary"
            onClick={dialogHandle}>
            Accept
          </Button>
        </div>
      </Dialog>
    );
  };

  const returnColumns = (item) => {
    return (
      <>
        <Card
          className={active === item.name ? classes.rootActive : classes.root}
          id={item.name}
          onClick={setClass}>
          <CardContent id={item.name} className={classes.content}>
            <div className={s.item_name_container}>
              <div className={s.item_name} id={item.name}>
                {item.name}
              </div>
              {isLoggedIn && props.user.role === 'admin' && (
                <div>
                  <IconButton
                    className={classes.editIcon}
                    size="small"
                    onClick={() => editItemHandle(item.name)}>
                    <EditIcon />
                  </IconButton>
                  <IconButton
                    className={classes.deleteIcon}
                    size="small"
                    onClick={() => deleteItemHandle(item.name)}>
                    <HighlightOffIcon />
                  </IconButton>
                </div>
              )}
            </div>

            <div id={item.name}>{item.category}</div>
            <div id={item.name}>{item.goal}</div>
            <div id={item.name} className={s.info}>
              <div id={item.name}>{item.interest}</div>
              <div className={s.date_active} id={item.name}>
                {item.dateActive}
              </div>
            </div>
          </CardContent>
        </Card>
      </>
    );
  };

  const filteredColumns = ['Active', 'Upcoming', 'Ended'];

  return (
    <div className={s.wrapper} id="container" onClick={setClass}>
      <div className={s.container}>
        {dialogForm()}
        <div className={s.content}>
          {filteredColumns.map((item) => {
            let columnName = '';
            if (item === 'Active') {
              columnName = 'Active';
              return (
                <>
                  <div className={s.column_container}>
                    <div className={s.title}>{columnName}</div>
                    <div className={s.column_content}>
                      {data
                        .filter((key) => key.column === columnName)
                        .map((item) => returnColumns(item))}
                    </div>
                  </div>
                </>
              );
            }
            if (item === 'Upcoming') {
              columnName = 'Upcoming';
              return (
                <>
                  <div className={s.column_container}>
                    <div className={s.title}>{columnName}</div>
                    <div className={s.column_content}>
                      {data
                        .filter((key) => key.column === columnName)
                        .map((item) => returnColumns(item))}
                    </div>
                  </div>
                </>
              );
            }
            if (item === 'Ended') {
              columnName = 'Ended';
              return (
                <>
                  <div className={s.column_container}>
                    <div className={s.title}>{columnName}</div>
                    <div className={s.column_content}>
                      {data
                        .filter((key) => key.column === columnName)
                        .map((item) => returnColumns(item))}
                    </div>
                  </div>
                </>
              );
            }
          })}
        </div>
        {isLoggedIn && props.user.role === 'admin' && (
          <div className={s.add_button}>
            Add new column
            <IconButton color="primary" size="medium" onClick={addItemHandle}>
              <AddBoxIcon fontSize="large" />
            </IconButton>
          </div>
        )}
        <div className={s.footer_container}>
          <div className={s.footer_row}>
            <div className={s.footer_content}>
              <div className={s.footer_title}>Name Company</div>
              <div className={s.footer_description}>
                ICO Drops is an independent ICO (Token Sale) database and is not affiliated with any
                ICO project or company. Our Interest Level does not constitute financial or
                investment advice.
              </div>
              <div className={s.footer_copyright}>© 2021 ICO Drops.</div>
            </div>
            <div className={s.footer_columnInfo}>
              <a href="#">Active ICO</a>
              <a href="#">Upcoming ICO</a>
              <a href="#">Ended ICO</a>
              <a href="#">Bounty List</a>
              <a href="#">SandBox </a>
            </div>
            <div className={s.footer_companyInfo}>
              <a href="#">Dropstab</a>
              <a href="#">Dropsearn</a>
              <a href="#">Portfolio</a>
              <a href="#">Ico calendar</a>
            </div>
            <div className={s.footer_mediaLink}>
              <a href="#">Email</a>
              <a href="#">Twitter</a>
              <a href="#">Telegram</a>
              <a href="#">Instagram</a>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default Home;
