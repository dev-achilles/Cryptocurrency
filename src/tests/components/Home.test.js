import React from 'react';
import renderer from 'react-test-renderer';
import Home from '../../components/Home';
import db from '../../db';

it('renders correctly', () => {
  const props = {
    user: {
      token: null,
      name: null,
      isLoggedIn: true,
      role: null,
      error: false,
    },
    homeData: db.cryptocurrency,
  };
  const tree = renderer.create(<Home user={props.user} homeData={props.homeData} />).toJSON();
  expect(tree).toMatchSnapshot();
});
