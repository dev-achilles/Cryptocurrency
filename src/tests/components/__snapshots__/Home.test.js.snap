// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="wrapper"
>
  <div
    className="container"
  >
    <div
      className="content"
    >
      <div
        className="column_container"
      >
        <div
          className="title"
        >
          Active
        </div>
        <div
          className="column_content"
        />
      </div>
      <div
        className="column_container"
      >
        <div
          className="title"
        >
          Upcoming
        </div>
        <div
          className="column_content"
        />
      </div>
      <div
        className="column_container"
      >
        <div
          className="title"
        >
          Ended
        </div>
        <div
          className="column_content"
        />
      </div>
    </div>
    <div
      className="add_button"
    >
      Add new column
      <button
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-2 10h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
            />
          </svg>
        </span>
      </button>
    </div>
    <div
      className="footer_container"
    >
      <div
        className="footer_row"
      >
        <div
          className="footer_content"
        >
          <div
            className="footer_title"
          >
            Name Company
          </div>
          <div
            className="footer_description"
          >
            ICO Drops is an independent ICO (Token Sale) database and is not affiliated with any ICO project or company. Our Interest Level does not constitute financial or investment advice.
          </div>
          <div
            className="footer_copyright"
          >
            Â© 2021 ICO Drops.
          </div>
        </div>
        <div
          className="footer_columnInfo"
        >
          <a
            href="#"
          >
            Active ICO
          </a>
          <a
            href="#"
          >
            Upcoming ICO
          </a>
          <a
            href="#"
          >
            Ended ICO
          </a>
          <a
            href="#"
          >
            Bounty List
          </a>
          <a
            href="#"
          >
            SandBox 
          </a>
        </div>
        <div
          className="footer_companyInfo"
        >
          <a
            href="#"
          >
            Dropstab
          </a>
          <a
            href="#"
          >
            Dropsearn
          </a>
          <a
            href="#"
          >
            Portfolio
          </a>
          <a
            href="#"
          >
            Ico calendar
          </a>
        </div>
        <div
          className="footer_mediaLink"
        >
          <a
            href="#"
          >
            Email
          </a>
          <a
            href="#"
          >
            Twitter
          </a>
          <a
            href="#"
          >
            Telegram
          </a>
          <a
            href="#"
          >
            Instagram
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
